src/util/records.py

```diff
@@ -310,6 +311,70 @@ def remove_asset(cr, name):
 # fmt:on
 
 
+def remove_action(cr, xml_id=None, action_id=None):
+    assert bool(xml_id) ^ bool(action_id)
+
+    action_model = None
+    if action_id:
+        cr.execute("SELECT type FROM ir_actions WHERE id=%s", [action_id])
+        [action_model] = cr.fetchone()
+    else:
+        action_id = ref(cr, xml_id)
+        module, _, name = xml_id.partition(".")
+        cr.execute("SELECT model FROM ir_model_data WHERE module=%s AND name=%s", [module, name])
+        [action_model] = cr.fetchone()
+        if action_model not in [ihn.model for ihn in for_each_inherit(cr, "ir.actions.actions")]:
+            raise ValueError(
+                "%r should point to a model inheriting from 'ir.actions.actions', not a %r" % (xml_id, action_model)
+            )
+
+    if version_gte("12.0"):
+        cr.execute(
+            """
+            SELECT name,
+                   model,
+                   on_delete
+              FROM ir_model_fields
+             WHERE relation = 'ir.actions.actions'
+               AND on_delete IN ( 'cascade', 'set null' )
+               AND ttype = 'many2one'
+            """
+        )
+
+        for column_name, model, on_delete in cr.fetchall():
+            model_table = table_of_model(cr, model)
+            if on_delete == "cascade":
+                query = format_query(cr, "DELETE FROM {} WHERE {} = %s", model_table, column_name)
+            else:
+                query = format_query(cr, "UPDATE {} SET {} = NULL WHERE {} = %s", model_table, column_name, column_name)
+            cr.execute(query, (action_id,))
+    else:
```

Comment:
We could do nothing instead.

Reply:
If it hasn't in all these years... we don't need to worry much now. That being said if we some case we can revisit this.